src/span.js

```diff
@@ -36,6 +36,7 @@ export default class Span {
     _tags: Array<Tag>;
     static _baggageHeaderCache: any;
```

Comment:
A relatively minor thing, but you might consider using a `Map` instead of an object for the baggage header cache. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_and_maps_compared